openid-4-verifiable-presentations-1_0.md

```diff
@@ -616,11 +622,322 @@ The Wallet then validates the request as specified in OAuth 2.0 [@RFC6749].
+be present otherwise. The value MUST be a claims path pointer that specifies the path to the claim
+within the Verifiable Credential, as defined in (#claims_path_pointer).
+
+`namespace`:
+: REQUIRED if the Credential Format is based on the mdoc format defined in ISO 18013-5; MUST NOT be present otherwise. 
+The value MUST be a string that specifies the namespace of the data element
+within the mdoc, e.g., `org.iso.18013.5.1`.
+
+`claim_name`:
+: REQUIRED if the Credential Format is based on mdoc format defined in ISO 18013-5; MUST NOT be present otherwise. 
+The value MUST be a string that specifies the data element identifier of the data element within the provided namespace
+in the mdoc, e.g., `first_name`.
+
+`values`:
+: OPTIONAL. An array of strings, integers or boolean values that specifies the expected values of the claim.
+If the `values` property is present, the Wallet SHOULD return the claim only if the
+type and value of the claim both match for at least one of the elements in the array. Details of the processing rules are defined in (#selecting_claims).
+
+### Selecting Claims and Credentials {#vp_query_lang_processing_rules}
+
+The following section describes the logic that applies for selecting claims 
+and for selecting credentials. 
+
+#### Selecting Claims {#selecting_claims}
+
+The following rules apply for selecting claims via `claims` and `claim_sets`:
+
+- If `claims` is not provided, the Verifier requests all claims existing
+  in the Credential.
+- If `claims` is provided, but `claim_sets` is not provided,
+  the Verifier requests all claims listed in `claims`.
+- Otherwise, the Verifier requests one combination of the claims listed in
+  `claim_sets`. The order of the options conveyed in the `claim_sets`
+  array expresses the Verifiers' preference for what is returned; the Wallet MUST return
+  the first option that it can satisfy. If the Wallet cannot satisfy any of the
+  options, it MUST NOT return any claims.
+
+When a Claims Query contains a restriction on the values of a claim, the Wallet
+SHOULD NOT return the claim if its value does not match at least one of the
+elements in `values` i.e., the claim should be treated the same as if it did not
+exist in the Credential. Implementing this restriction may not be possible in
+all cases, for example, if the Wallet does not have access to the claim value
+before presentation or user consent or if another component routing
+the request to the Wallet does not have access to the claim value. Therefore, Verifiers
+must treat restrictions expressed using `values` as a best-effort way to improve
+user privacy, but MUST NOT rely on it for security checks.
+
+The purpose of the `claim_sets` syntax is to provide a way for a 
+verifier to describe alternative ways a given credential can satisfy the request. 
+The array ordering expresses the Verifier's preference for how the request must 
+be fulfilled. The first element in the array is the most preferred and the last 
```

Comment:
```suggestion
The array ordering expresses the Verifier's preference for how to fulfill the request. The first element in the array is the most preferred and the last 
```
to address David's comment above

Reply:
This a different sentence. Mine is lines line 791 to 793

`Otherwise, the Verifier requests all of the credential set queries in the credential_sets array with required evaluated as true to be returned at a minimum and optionally any of the credential set queries with required evaluated as false.`